services/app-web/src/components/Form/ErrorSummary/ErrorSummary.tsx

```diff
@@ -59,6 +72,11 @@ export type ErrorSummaryProps = {
  *
  * This component relies on the errors object returned by Formik, so it should generally
  * rendered inside of a Formik form context.
+ * 
+ * By default, keys in the errors property are assumed to correspond to the name attribute
+ * of the relevant form inputs. By passing a key that begins with a number sign ("#"),
+ * you can have it treated as an ID. This is useful when you want to focus something that
+ * doesn't have a name attribute when an error message is clicked.
```

Comment:
This is the main change: `errorKey`s that start with `#` are treated as IDs instead of values of `name` attributes.

There's some cleanup that would be nice to do- we have places where the names and `id`s don't match but probably could, and this means that some of our links to those elements don't go anywhere. It's not an issue since we handle the focus in JS, but adding more consistency over time would be nice.

Reply:
Thanks for adding the comment here too I think that's clear.